core/engine/src/vm/opcode/args.rs

```diff
@@ -807,3 +939,654 @@ impl Argument for (VaryingOperand, ThinVec<u32>) {
         ((first.into(), rest), pos + 6 + total_len * 4)
     }
 }
+
+#[cfg(test)]
+mod test {
+
+  use std::vec;
+  use thin_vec::thin_vec;
+  use super::*;
+
+  fn test_write_arg<T: Argument>(test_data: T) -> Vec<u8> {
+    let mut v: Vec<u8> = Vec::new();
+    test_data.encode(&mut v);
+    return v
+  }
+
+  // This can fail depending on Argument/opcode combination given (too short)
+  fn test_read_arg<T: Argument>(opcode: Box<[u8]>, pos: usize) -> (T, usize) {
+    let r: &[u8] = &opcode;
+    T::decode(r, pos)
+  }
+
+
+  //
+  // VariableOperand
+  //
+  #[test]
+  fn arg_vop_u8() {
+    test_write_arg(VaryingOperand{value: u8::MAX as u32});
```

Comment:
We can use [From<T>](https://github.com/boa-dev/boa/blob/dddc98ffe279f21e7ed79cfd6d592389181c5f5e/core/engine/src/vm/opcode/mod.rs#L235-L269) impl on `VaryingOperand`
```suggestion
    test_write_arg(VaryingOperand::from(u8::MAX));
```